### PR TITLE
Add "Restart app" button to Developer Settings

### PR DESCRIPTION
**Pull request by DmitriyZaitsev**
... using [Process Phoenix](https://github.com/JakeWharton/ProcessPhoenix). Enhancement #58

### CODE REVIEW

**Comment by vanniktech** [comment]
:+1:

File: app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java [hunk 1]
```diff
@@ -105,6 +108,12 @@ void onHttpLoggingLevelChanged(int position) {
         presenter.changeHttpLoggingLevel(((HttpLoggingLevel) httpLoggingLevelSpinner.getItemAtPosition(position)).loggingLevel);
     }
 
+    @OnClick(R.id.developer_settings_restart_app)
+    void onRestartAppClick() {
+        final FragmentActivity activity = getActivity();
+        ProcessPhoenix.triggerRebirth(activity, new Intent(activity, activity.getClass()));
```
Comment (lenguyenthanh) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: Should we delegate this job to Presenter

Comment (DmitriyZaitsev) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: @lenguyenthanh, I considered this, but
• current presenter knows nothing about Android SDK-dependent types (context, view etc).
• I followed existing approach - to work  with context within this fragment like it's done e.g. in [`showLog()`](https://github.com/artem-zinnatullin/qualitymatters/blob/d101253d7e4ad58ea2bd03bde953e29700253fd4/app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java#L210)

Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/fragments/DeveloperSettingsFragment.java]: Well, yeah, though just as note: we can wrap ProcessPhoenix into separate class and invoke it from the presenter, but I'm ok with current solution

File: app/src/debug/res/layout/fragment_developer_settings.xml [hunk 2]
```diff
@@ -175,6 +175,16 @@
 
         </LinearLayout>
 
+        <Button
+            android:id="@+id/developer_settings_restart_app"
```
Comment (artem-zinnatullin) [inline app/src/debug/res/layout/fragment_developer_settings.xml]: nit: can you please add `_button` to the end of id? Just easier to work with such ids in future :)

**Comment by artem-zinnatullin** [comment]
:+1:

File: app/src/debug/res/layout/fragment_developer_settings.xml (hunk 2 above)
Comment (DmitriyZaitsev) [inline app/src/debug/res/layout/fragment_developer_settings.xml]: done :)

**Comment by vanniktech** [comment]
:+1:

**Comment by artem-zinnatullin** [comment]
Thanks!

**Pull request merged**